processing/src/main/java/org/apache/druid/segment/column/ColumnCapabilities.java

```diff
@@ -38,6 +38,7 @@
   boolean hasSpatialIndexes();
   Capable hasMultipleValues();
   boolean isFilterable();
+  Capable isNullable();
```

Comment:
Broad comment: "hasNulls" would be a better name for this property, because it's meant to mean whether or not the column really has nulls in it right now.

"isNullable" will make people think of the SQL sense of the term, which is a different concept: it means a column could have nulls potentially, even if it doesn't right now.

Reply:
I agree, and have renamed this to `hasNulls`, but haven't changed the segment metadata `ColumnAnalysis` yet... 

I view segment metadata as the computed equivalent of an SQL schema, and have exposed in with the intention of populating the `IS_NULLABLE` column in the information schema columns table, and access via `DruidSchema` will allow the SQL planner to make better plans and skip null checking and such, so I thought it still appropriate to leave this as `isNullable`. I don't feel super strongly about this though, so if you disagree I can change it.

Reply:
I do feel strongly that it shouldn't be called `isNullable`. I think we should avoid the word "nullable" in core Druid to refer to this concept, because one day, we might want to introduce the SQL-style concept of nullable vs non-nullable columns. That concept is more of a definition thing than a reality thing, and will be confusing to introduce to core if we have the reality-thing using the word "nullable".

Also, if we did introduce it to core, we'd probably want to report both (a column that is nullable table-wide might not actually have nulls in a specific segment, and that's useful to know).